pcicrawler/ocp_diag.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+
+"""
+Copyright (c) Meta Platforms, Inc. and affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+import json
+import socket
+import pkg_resources
+import typing as ty
+import ocptv.output as tv
+from pci_lib import PCIDevice
+from abc import abstractmethod
+
+
+class OCPTestStep:
+    def __init__(self,
+                    device: PCIDevice,
+                    expected_result: str,
+                    ocp_run: tv.TestRun):
+        self._device = device
+        self._expected_result = expected_result
+        self._ocp_run = ocp_run
+        self._error_messages = []
+
+    @property
+    @abstractmethod
+    def _test_name(self):
```

Comment:
in case this isnt as clear; this def property does _not_ match the classvar youve set on the inheriting classes. they are different things, it's only a consequence of MRO that accessing said name resolves to the classvar

in order to impl this on the inheritors  you need to impl the property, eg
```python
class CheckLocation(OCPTestStep):
    @property
    def _test_name(self):
        return "pci_location_check"
```

alternative is to use a typing check to show that the usage of `self._test_name` here is to be sourced from a classvar in inheritors (which is probably easier and shorter than the first option)